src/pages/host/VansList.jsx

```diff
@@ -1,23 +1,37 @@
 import Van from "../../components/cards/Van";
 import { Heading2 } from "../../components/ui/Typography";
-import { useLoaderData } from "react-router-dom";
+import { defer, useLoaderData, Await } from "react-router-dom";
 import { getHost } from "../../api";
 import requireAuth from "../../../utils";
+import { Suspense } from "react"
 export const loader = async ({ request }) => {
   await requireAuth(request);
-  return getHost();
+  return defer({ vans: getHost() })
 };
 
 const VansList = () => {
   const vansList = useLoaderData();
   return (
     <div className="bg-[#fff7ed] w-full text-center">
       <Heading2>Your listed Vans</Heading2>
-
       <div className="w-full justify-center flex-col items-center flex gap-y-5 py-5">
-        {vansList.map((van) => (
-          <Van {...van} key={van.id} />
-        ))}
+        <Suspense fallback={<div className="my-10">
+          <Heading2>Loading vans...</Heading2>
+        </div>}>
+          <Await resolve={vansList.vans}>
+            {
+              (vans) => {
+                return (
+                  vans.map((van) => (
```

Comment:
I would advise you to make this portion of code shorter by using the implicit return. Like this:
```jsx
{
    vans => vans.map()
}
```